sirepo/pkcli/db.py

```diff
@@ -34,3 +34,86 @@ def _inc(m):
                 break
             with open(str(fn), 'w') as f:
                 f.write(t)
+
+
+def seed_supervisor_state(db_dir, sbatch_poll_secs):
+    from pykern import pkio
+    from pykern.pkcollections import PKDict, json_load_any
+    from pykern.pkdebug import pkdp
+    from sirepo import job
+    from sirepo import simulation_db
+    from sirepo import util
+    from sirepo import sim_data
+    import sirepo.template
+    import os
+
+    db_dir = pkio.py_path(db_dir)
+    pkio.mkdir_parent(db_dir)
+
+    _NEXT_REQUEST_SECONDS = PKDict({
+        job.PARALLEL: 2,
+        job.SBATCH: int(sbatch_poll_secs),
+        job.SEQUENTIAL: 1,
+    })
+
+    def _add_parallel_status(in_json, sim_data, run_dir, data):
+        t = sirepo.template.import_module(data.simulationType)
+        data.parallelStatus = PKDict(
+           t.background_percent_complete(
+               sim_data.parse_model(in_json),
+               run_dir,
+               False,
+           )
+        )
+
+    def _create_supervisor_state_file(uid, run_dir):
+        i = json_load_any(
+            run_dir.join(
+                sirepo.template.template_common.INPUT_BASE_NAME +
+                simulation_db.JSON_SUFFIX
+            )
+        )
+        c = sim_data.get_class(i.simulationType)
+        p = c.is_parallel(i)
+        j = job.PARALLEL if p else job.SEQUENTIAL
+        s, t = _read_status_file(run_dir)
```

Comment:
pass `d` to read_status file, and have it fill in values instead of using intermediate parameters as is done in _add_parallel_status.